backend/services/onshape_client.py

```
@@ -0,0 +1,223 @@
+            or "forbidden" in error_str
+        ):
+            reason = "403 Forbidden"
+        else:
+            reason = str(e)
+
+        # Return structured 400 response with computed IDs
+        raise ClientException(
+            _build_preflight_error_response(
+                reason, document_id, wvm, wvmid, element_id, mode
+            )
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)